pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -323,7 +333,12 @@ func (r ReplicaSetReconciler) buildAutomationConfigConfigMap(mdb mdbv1.MongoDB)
 	if err != nil {
 		return corev1.ConfigMap{}, fmt.Errorf("error reading version manifest from disk: %+v", err)
 	}
-	ac := buildAutomationConfig(mdb, manifest.BuildsForVersion(mdb.Spec.Version))
+	currentCm := &corev1.ConfigMap{}
+	currentAc := automationconfig.AutomationConfig{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, currentCm); err == nil {
```

Comment:
A good practice around error handling/indentation in go is to make sure that the happy path of the code is always at the left most side of the function. This basically translates to check for `err != nil` rather than `err == nil`

I think this is a good pattern to follow, as it reduces the scope of where we need to look to see what a function is doing when reading it.

so basically

```golang
if err := GetConfigMap(); err != nil {
    return err
}
// do thing with configmap
```

Reply:
I see your point. In this case we would still need to do things with configmap inside a branch since not all errors from the get of the config map need to terminate the function (is the error is IsNotFound we are ok with that). So it would looke like this:

```golang
if err := r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, currentCm); err != nil {
	if !errors.IsNotFound(err) {
		return corev1.ConfigMap{}, err
	}
} else {
	if err := json.Unmarshal([]byte(currentCm.Data[AutomationConfigKey]), &currentAc); err != nil {
		return corev1.ConfigMap{}, err
	}
}
```